pkg/apis/core/v1beta1/types_seed.go

```diff
@@ -270,8 +270,23 @@ type SeedSettings struct {
 
 // SeedSettingExcessCapacityReservation controls the excess capacity reservation for shoot control planes in the seed.
 type SeedSettingExcessCapacityReservation struct {
-	// Enabled controls whether the excess capacity reservation should be enabled.
+	// Enabled controls whether the default excess capacity reservation should be enabled.
+	// Deprecated: This field is deprecated and will be removed in a future version of Gardener. Please use `Configs` instead.
 	Enabled bool `json:"enabled" protobuf:"bytes,1,opt,name=enabled"`
+	// Configs configures excess capacity reservation deployments for shoot control planes in the seed.
+	Configs []SeedSettingExcessCapacityReservationConfig `json:"configs,omitempty" protobuf:"bytes,2,rep,name=configs"`
+}
+
+// SeedSettingExcessCapacityReservationConfig configures excess capacity reservation deployments for shoot control planes in the seed.
+type SeedSettingExcessCapacityReservationConfig struct {
+	// Resources specify the resource requests and limits of the excess-capacity-reservation pod.
+	Resources corev1.ResourceList `json:"resources,omitempty" protobuf:"bytes,1,rep,name=resources"`
+	// NodeSelector specifies the node where the excess-capacity-reservation pod should run.
+	// +optional
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty" protobuf:"bytes,2,opt,name=nodeSelector"`
+	// Tolerations specify the tolerations for the the excess-capacity-reservation pod.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty" protobuf:"bytes,3,rep,name=tolerations"`
```

Comment:
You could validate the tolerations with https://github.com/kubernetes/kubernetes/blob/e2afa175e4077d767745246662170acd86affeaf/pkg/apis/core/validation/validation.go#L3722-L3763.

You could check whether the other fields need validation as well.

Reply:
I added some validations. Apparently, it is the first time we are reusing validations from kubernetes core.
Additionally, we were reusing kubernetes external datatypes only in our internal data types so far. These are the first kubernetes internal data types.